util/util.go

```diff
@@ -138,6 +142,96 @@ func AbiBigIntTobytes(src *big.Int) []byte {
 	return res
 }
 
+func AbiDeployArgsTobytes(src []*consensus.Deploy_Arg) []byte {
+	res := make([]byte, 4)
+	binary.LittleEndian.PutUint32(res, uint32(len(src)))
+
+	for _, deployArg := range src {
+		bytes := AbiDeployArgTobytes(deployArg.GetValue())
+		lenBytes := make([]byte, 4)
+		binary.LittleEndian.PutUint32(lenBytes, uint32(len(bytes)))
+		res = append(res, lenBytes...)
+		res = append(res, bytes...)
+	}
+
+	return res
+}
+
+func AbiDeployArgTobytes(src *consensus.Deploy_Arg_Value) (res []byte) {
+	var data []byte
+	switch src.GetValue().(type) {
+	case *consensus.Deploy_Arg_Value_OptionalValue:
+		data = AbiOptionToBytes(AbiDeployArgTobytes(src.GetOptionalValue()))
+	case *consensus.Deploy_Arg_Value_BytesValue:
+		data = src.GetBytesValue()
+	case *consensus.Deploy_Arg_Value_IntValue:
+		data = AbiUint32ToBytes(uint32(src.GetIntValue()))
+	case *consensus.Deploy_Arg_Value_IntList:
+		// TODO : Need to test
+		intValues := src.GetIntList().GetValues()
+		for _, value := range intValues {
+			data = append(data, AbiUint32ToBytes(uint32(value))...)
+		}
+	case *consensus.Deploy_Arg_Value_StringValue:
+		data = AbiStringToBytes(src.GetStringValue())
+	case *consensus.Deploy_Arg_Value_StringList:
+		// TODO : Need to test
+		strValues := src.GetStringList().GetValues()
+		for _, value := range strValues {
+			data = append(data, AbiStringToBytes(value)...)
+		}
+	case *consensus.Deploy_Arg_Value_LongValue:
+		data = AbiUint64ToBytes(uint64(src.GetLongValue()))
+	case *consensus.Deploy_Arg_Value_BigInt:
+		// TODO : ParseUint didn't support 512.. need to change more.
+		bitWidth := uint32(64)
+		if src.GetBigInt().GetBitWidth() < 64 {
+			bitWidth = src.GetBigInt().GetBitWidth()
+		}
+		val, err := strconv.ParseUint(src.GetBigInt().GetValue(), 10, int(bitWidth))
+		if err != nil {
+			return []byte{}
```

Comment:
Why don't we propagate the error?

Reply:
https://github.com/hdac-io/casperlabs-ee-grpc-go-util/pull/9/commits/d7bced484050cbeedfe9b462107a9b460d946c51